### PR TITLE
[Java] fix UT segmentation fault on exit

### PR DESCRIPTION
**Pull request by kfstorm**
## Why are these changes needed?

Ray's signal handler may conflict with the signal handler set by JVM. Here we avoid calling `InstallFailureSignalHandler` during the initialization of a Java worker and calling `UninstallSignalAction` during the exit of JVM.

It's possible to fix this by chaining signal handlers, as described here: https://docs.oracle.com/javase/9/troubleshoot/handle-signals-and-exceptions.htm#GUID-CB49A2A7-2A9F-4C18-948F-6D4A96FF688D. But it's not easy and I'm not going to drive it in this PR.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Review by raulchen — COMMENTED** [review]
What about just using a flag to save whether `InstallFailureSignalHandler` was called? When when shutting down the log, we just check the flag and don't need to pass in that parameter

File: src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc [hunk 1]
```diff
@@ -108,12 +108,15 @@ JNIEXPORT void JNICALL Java_org_ray_runtime_RayNativeRuntime_nativeSetup(JNIEnv
                                                                          jstring logDir) {
   std::string log_dir = JavaStringToNativeString(env, logDir);
   ray::RayLog::StartRayLog("java_worker", ray::RayLogLevel::INFO, log_dir);
-  // TODO (kfstorm): If we add InstallFailureSignalHandler here, Java test may crash.
+  // TODO (kfstorm): We can't InstallFailureSignalHandler here, because JVM already
+  // installed its own signale handler.
```
Comment (zhijunfu) [inline src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc]: good to also mention "It's possible to fix this by chaining signal handlers but it's not easy, for more details, see ... "

**Review by zhijunfu — APPROVED** [review]

**Comment by kfstorm** [comment]
> What about just using a flag to save whether `InstallFailureSignalHandler` was called? When when shutting down the log, we just check the flag and don't need to pass in that parameter

Nice suggestion! Done.

File: src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc (hunk 1 above)
Comment (kfstorm) [reply src/ray/core_worker/lib/java/org_ray_runtime_RayNativeRuntime.cc]: Done.

**Review by raulchen — APPROVED** [review]

**Pull request merged**